src/launchpad/analyzers/macho_parser.py

```diff
@@ -0,0 +1,324 @@
+                return (code_sig.data_offset, code_sig.data_size)
+        return None
+
+    def _cpu_type_to_string(self, cpu_type: int) -> Optional[str]:
+        """Convert LIEF CPU type to string representation."""
+        # Common CPU types from Mach-O
+        cpu_types = {
+            0x0000000C: "arm",  # ARM
+            0x0100000C: "arm64",  # ARM64
+            0x00000007: "x86",  # i386
+            0x01000007: "x86_64",  # x86_64
+        }
+        return cpu_types.get(cpu_type)
+
+    def get_section_content(self, section_name: str) -> bytes | None:
+        """Get raw bytes content of a specific section.
+
+        Args:
+            section_name: Name of the section to retrieve
+
+        Returns:
+            Raw bytes of the section content, or None if section not found
+        """
+        try:
+            if not hasattr(self.binary, "sections"):
+                return None
+
+            for section in self.binary.sections:
+                if hasattr(section, "name") and section.name == section_name:
+                    if hasattr(section, "content"):
+                        content = section.content
+                        return bytes(content)
+
+            logger.debug(f"Section {section_name} not found")
+            return None
+
+        except Exception as e:
+            logger.debug(f"Failed to get section content for {section_name}: {e}")
+            return None
+
+    def get_section_bytes_at_offset(self, section_name: str, offset: int, size: int) -> bytes | None:
+        """Get specific bytes from a section at a given offset.
+
+        Args:
+            section_name: Name of the section
+            offset: Offset within the section
+            size: Number of bytes to read
+
+        Returns:
+            Raw bytes at the specified offset, or None if not found
+        """
+        try:
+            content = self.get_section_content(section_name)
+            if content is None:
+                return None
+
+            if offset + size > len(content):
+                logger.warning(f"Requested range {offset}:{offset+size} exceeds section size {len(content)}")
+                return None
+
+            return content[offset : offset + size]
+
+        except Exception as e:
+            logger.debug(f"Failed to get section bytes at offset for {section_name}: {e}")
+            return None
+
+    def search_bytes_in_section(self, section_name: str, pattern: bytes) -> list[int]:
+        """Search for a byte pattern within a section.
+
+        Args:
+            section_name: Name of the section to search in
+            pattern: Byte pattern to search for
+
+        Returns:
+            List of offsets where the pattern was found
+        """
+        try:
+            content = self.get_section_content(section_name)
+            if content is None:
+                return []
+
+            offsets = []
+            start = 0
+            while True:
```

Comment:
while can be technically fine, `while True` seems like a code smell. Should we maybe add a maximum # of iterations?